### ISSUE TITLE
[Enhancement]: Disallow creating order containing permanently deleted items

### ISSUE THREAD

**Issue by samiuelson** [woothemes/woocommerce#50698]
#### Context 
The request below comes from the Woo Mobile team working on in-person payments and Point of Sale.

#### What
We'd like to request changing [order creation](https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order) endpoint to return an error in case there are non-existent order items sent in the request. 

##### Current behavior
The endpoint returns success if the order creation endpoint is called with a non-existent item (e.g. permanently deleted). However, in the response, the item totals are set to 0.00. 

In the Woo Mobile app, this leads to a misleading state in which the order totals presented to the merchant are not in line with the items visible in the cart:
<img src=https://github.com/user-attachments/assets/d9930b88-072b-43bd-9455-9cb8d7d797ee width="560"/>
In the screenshot above, the item in the cart (on the left) has been permanently deleted. In the right pane, the totals reflect response of the order creation endpoint.

##### Desired behavior
We would benefit from the endpoint returning an error if the request contains non-existing order items.

We could prevent merchants from charging clients incorrect amounts and would be able to detect and inform them about the problem before starting in-person payment collection flow.

### Describe alternatives you've considered

Unlike the wp-admin web interface, in the mobile apps, because of performance implications, we are not able to reload products each time we start the order creation flow. We need to maintain a local cache of the products, however, we are not getting any remote notification about a product being permanently deleted, allowing us to invalidate the cache. 

Receiving a push notification, informing the app to invalidate the products cache would work as well. However, it's a greater effort. That said we suggest modifying the response of the endpoint to return an error, making it more predictable from the point of view of the mobile app.

**Comment by samiuelson** [comment]
I'm curious what are your thoughts about the proposal, @coreymckrill!

**Comment by coreymckrill** [comment]
@samiuelson We decided to re-label this as a bug, because it seems unlikely that the behavior of allowing the addition of a line item with an invalid product ID is intentional.

Being mindful of back compatibility, we probably shouldn't prevent a line item from being added just because it doesn't have a valid product. However, in the case that a product ID (or variation ID) is included in the request, it seems reasonable for us to run validation on the ID and respond with an error if the validation fails.

In a separate issue, it might be interesting to explore what it would take to set up push notifications for things like product deletion. Don't we already have PNs for things like new orders?

**Comment by samrat-biswas-git** [comment]
Reviewed and confirmed this is still valid. Marking as ready for engineering.

Current evidence: Trunk still allows REST order item creation to proceed when `wc_get_product()` returns false for a deleted product reference. The controller path validates some item data but does not reject permanently deleted product IDs, so the reported bad order-creation behavior remains possible.
